client/wsclient.go

```diff
@@ -85,7 +87,41 @@ func (c *wsClient) Stop(ctx context.Context) error {
 	conn := c.conn
 	c.connMutex.RUnlock()
 
+	ticker := time.NewTicker(50 * time.Millisecond)
+
+	// Wait for all remaining messages to be sent. Continuing to attempt
+	// to send messages after calling Stop will eventually result in lost
+	// messages.
+	select {
```

Comment:
I am confused by what these select and cases do.

Reply:
The goal of this select is to allow the sender to deliver any queued messages by polling the sender for whether it has any messages remaining. Currently in the Supervisor we call `SetHealth` immediately before calling `Stop`, and the lack of delay between the two doesn't allow time for the health message to be sent and produces a `websocket: close sent` error.

This was the most obvious solution to the issue, but I'll take another look to see whether there are other viable approaches.